File: YangDoPlatform/app/src/main/java/se/jbnu/yangdoplatform/chat/ChatAdapter.java
package se.jbnu.yangdoplatform.chat;

import android.os.Build;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.RequiresApi;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import se.jbnu.yangdoplatform.R;

public class ChatAdapter extends RecyclerView.Adapter<ChatAdapter.MyViewHolder> {
    private List<Chat> chatList;
    private String name;

    public ChatAdapter(List<Chat> chatData, String name){
        chatList = chatData;
        this.name = name;
    }
    public static class MyViewHolder extends RecyclerView.ViewHolder{
        public TextView nameText;
        public TextView msgText;
        public View rootView;
        public LinearLayout msgLinear;
        public MyViewHolder(@NonNull View itemView) {
            super(itemView);
            msgLinear = itemView.findViewById(R.id.msgLinear);
            nameText = itemView.findViewById(R.id.nameText);
            msgText = itemView.findViewById(R.id.msgText);
            rootView = itemView;

            itemView.setEnabled(true);
            itemView.setClickable(true);
        }
    }
    @NonNull
    @Override
    public ChatAdapter.MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LinearLayout linearLayout = (LinearLayout) LayoutInflater.from(parent.getContext()).inflate(R.layout.item_msg,parent,false);

        MyViewHolder myViewHolder = new MyViewHolder(linearLayout);

        return myViewHolder;
    }
    @RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN_MR1)
    @Override
    public void onBindViewHolder(@NonNull ChatAdapter.MyViewHolder holder, int position) {

        Chat chat = chatList.get(position);

        holder.nameText.setText(chat.getName());
        holder.msgText.setText(chat.getMsg());
        if(chat.getName().equals(this.name)){
            holder.nameText.setTextAlignment(View.TEXT_ALIGNMENT_VIEW_END);
            holder.msgText.setTextAlignment(View.TEXT_ALIGNMENT_VIEW_END);

            holder.msgLinear.setGravity(Gravity.RIGHT);
        } else {
            holder.nameText.setTextAlignment(View.TEXT_ALIGNMENT_VIEW_START);
            holder.msgText.setTextAlignment(View.TEXT_ALIGNMENT_VIEW_START);

            holder.msgLinear.setGravity(Gravity.LEFT);
        }
    }

    @Override
    public int getItemCount() {
        return chatList == null ? 0: chatList.size();
    }



    public void addChat(Chat chat){
        chatList.add(chat);
        notifyItemInserted(chatList.size()-1);
    }
}

File: YangDo_Platform/app/src/main/java/se/jbnu/yangdoplatform/chat/ChatAdapter.kt
package se.jbnu.yangdoplatform.chat

import android.os.Build
import android.view.Gravity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView
import androidx.annotation.RequiresApi
import androidx.recyclerview.widget.RecyclerView
import se.jbnu.yangdoplatform.R
import se.jbnu.yangdoplatform.chat.ChatAdapter.MyViewHolder

class ChatAdapter(private val chatList: MutableList<Chat>?, private val name: String) : RecyclerView.Adapter<MyViewHolder>() {
    class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var nameText: TextView
        var msgText: TextView
        var rootView: View
        var msgLinear: LinearLayout

        init {
            msgLinear = itemView.findViewById(R.id.msgLinear)
            nameText = itemView.findViewById(R.id.nameText)
            msgText = itemView.findViewById(R.id.msgText)
            rootView = itemView
            itemView.isEnabled = true
            itemView.isClickable = true
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val linearLayout = LayoutInflater.from(parent.context).inflate(R.layout.item_msg, parent, false) as LinearLayout
        return MyViewHolder(linearLayout)
    }

    @RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN_MR1)
    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val chat = chatList!![position]
        holder.nameText.text = chat.name
        holder.msgText.text = chat.msg
        if (chat.name == name) {
            holder.nameText.textAlignment = View.TEXT_ALIGNMENT_VIEW_END
            holder.msgText.textAlignment = View.TEXT_ALIGNMENT_VIEW_END
            holder.msgLinear.gravity = Gravity.RIGHT
        } else {
            holder.nameText.textAlignment = View.TEXT_ALIGNMENT_VIEW_START
            holder.msgText.textAlignment = View.TEXT_ALIGNMENT_VIEW_START
            holder.msgLinear.gravity = Gravity.LEFT
        }
    }

    override fun getItemCount(): Int {
        return chatList?.size ?: 0
    }

    fun addChat(chat: Chat) {
        chatList!!.add(chat)
        notifyItemInserted(chatList.size - 1)
    }
}